client/my-sites/themes/thanks-modal.jsx

```diff
@@ -24,9 +24,10 @@ import {
 	isActivatingTheme,
 	hasActivatedTheme,
 	isWpcomTheme,
+	shouldEditThemeWithGutenberg as isThemeGutenbergFirst,
```

Comment:
I personally find this name switcheroo a bit odd.
I guess we are changing the name twice in order to use both the `shouldEditThemeWithGutenberg` name and the props destructuring.

Readability-wise it's the right choice, but then I wonder: what if we turned this upside down and had the actual selector called `isThemeGutenbergFirst` in the first place?

E.g. (pseudo code with an unbelievable amount of errors, inconsistencies, and whatnot 😄)
```jsx
import { isThemeGutenbergFirst } from 'state/selectors/is-theme-gutenberg-first'; // see the other comment I've left :P

// ...

getEditSiteLabel = () => {
  const { shouldEditThemeWithGutenberg } = this.props;
  return shouldEditThemeWithGutenberg ? 'Edit Homepage' : 'Customize site' }
};

// ...

export default connect( state => ( {
  shouldEditThemeWithGutenberg: isThemeGutenbergFirst( state, currentThemeId )
} ) )( ThanksModal );
```

Reply:
Ok, I can see now that when the selector is used anywhere else, it makes more sense as called `shoutlEditTheme--` instead of `isTheme--`.

I'd still try to keep the name switcheroo as simple as possible though, to avoid confusion.

What do you think of being super clear here and going with `shouldEditHomepageWithGutenberg`?

```jsx
import { shouldEditHomepageWithGutenberg } from 'state/selectors/should-edit-homepage-with-gutenberg';

export default connect( state => ( {
  shouldEditThemeWithGutenberg: shouldEditHomepageWithGutenberg( state, currentThemeId )
} ) )( ThanksModal );
```

Reply:
> in order to use both the shouldEditThemeWithGutenberg name and the props destructuring

yep!

> shouldEditHomepageWithGutenberg

Hm. I don't like this name because it implies the setting is related to the homepage--when it is actually related to the theme only. I.e. The theme supports gutenberg styles, and so we want to open gutenberg instead. I also struggled with this one a little bit. "shouldEditTheme" doesn't make sense in that we aren't editing the theme... but it seemed better, especially since it's specifically calculating whether certain properties exist on the theme.

Reply:
maybe `isThemeGutenbergFirst` does make more sense overall